stumpy/floss.py

```diff
@@ -514,12 +551,32 @@ def __init__(
                 - core.z_norm(right_nn, 1),
                 axis=1,
             )
+            nn_subseq_isconstant = self._T_subseq_isconstant[
+                self._mp[:, 3].astype(np.int64)
+            ]
+            # subseq and its nn are both constant
+            mask = self._T_subseq_isconstant & nn_subseq_isconstant
+            self._mp[mask, 0] = 0
+
+            # Only the subseq OR its nn is constant but not both
+            mask = np.logical_xor(self._T_subseq_isconstant, nn_subseq_isconstant)
+            self._mp[mask, 0] = np.sqrt(m)
+
         else:
             self._mp[:, 0] = np.linalg.norm(
                 core.rolling_window(self._T, self._m) - right_nn,
                 axis=1,
                 ord=self._p,
             )
```

Comment:
Below I have added the comment based on what you mentioned before. I expanded it a little. But now I think it may be too verbose. If you feel the same way, I can prune it.

Reply:
Yes, it is too long. How about:

> Note that a negative matrix profile index (e.g. `-1`) represents a null index. However, in numpy, negative indices are interpreted as counting from the end of the array. To resolve this, we temporarily replace a negative index with a self index. So, if subsequence `S_i` has no valid right nearest neighbor, then `self._mp[i, 3]` will be set to `i` and its corresponding matrix profile distance is set to 0.0. This temporary resolution is later resolved (by setting them back to `-1` and `np.inf`) in a post-processing step.

Reply:
I kinda like that as it provides just enough of a reminder that is clear and concise

Reply:
> clear and (mostly) concise

Right! I will revise the comment. Thank you for the feedback :)